### ISSUE TITLE
Publishing Workflow Upgrade - Phase 2: gh-pages History Cleanup

### ISSUE THREAD

**Issue by mmcky** [QuantEcon/meta#261]
## Overview

This issue tracks the implementation of Phase 2 publishing workflow upgrades across QuantEcon lecture repositories. The upgrade performs a one-time cleanup of gh-pages branch history to dramatically reduce repository sizes.

**Prerequisite**: Phase 1 (HTML Archive to Release Assets) is complete - see [#253](https://github.com/QuantEcon/meta/issues/253)

## Objective

Remove accumulated deployment commits from gh-pages branches to reduce repository sizes by ~96%.

## Expected Results

| Metric | Before | After | Improvement |
|--------|--------|-------|-------------|
| Repository size | ~4 GB | ~100 MB | 96% reduction |
| Fresh clone time | 5-10 min | ~30 sec | 90% faster |
| gh-pages commits | 100-200+ | 1 | 99% fewer |
| Bandwidth per clone | ~4 GB | ~100 MB | 97% less |

### 1. Create Backup Branch
```bash
git fetch origin gh-pages
git branch gh-pages-backup origin/gh-pages
```

### 2. Create Fresh Orphan Branch
```bash
git checkout gh-pages
git checkout --orphan gh-pages-new
git add -A
git commit -m "Fresh gh-pages deployment (history cleanup to reduce repo size)"
```

### 3. Replace Old gh-pages
```bash
git branch -D gh-pages
git branch -m gh-pages
git push origin gh-pages --force
```

### 4. Verify
```bash
# Should show 1 commit
git rev-list --count origin/gh-pages

# Verify website still works
# Visit the site URL
```

### Active Lecture Repositories

| Repository | Site URL | Status | Notes |
|-----------|----------|--------|-------|
| lecture-python.myst | python.quantecon.org | ⏳ Pending | |
| lecture-python-programming.myst | python-programming.quantecon.org | ⏳ Pending | |
| lecture-python-advanced.myst | python-advanced.quantecon.org | ⏳ Pending | |
| lecture-python-intro | intro.quantecon.org | ⏳ Pending | |
| lecture-jax | jax.quantecon.org | ⏳ Pending | |
| lecture-intro.zh-cn | intro-zh-cn.quantecon.org | ⏳ Pending | |
| lecture-python-programming.fa | - | ⏳ Pending | |
| lecture-datascience.myst | datascience.quantecon.org | ⏳ Pending | |
| lecture-julia.myst | julia.quantecon.org | ⏳ Pending | |
| lecture-python.zh-cn | - | ⏳ Pending | |

### Legacy Repositories (Skipped)

The following repositories are scheduled for archival and will not receive this upgrade:
- `lecture-dle`
- `lecture-dps`
- `lecture-dynamics`
- `lecture-econometrics-machine-learning`
- `lecture-eqm`
- `lecture-stats`
- `lecture-tools-techniques`

## Rollback Procedure

If issues occur after cleanup:
```bash
git push origin gh-pages-backup:gh-pages --force
```

## Related

- **Phase 1 Issue**: [#253](https://github.com/QuantEcon/meta/issues/253) (Complete)
- **Original Implementation Plan**: [lecture-python.myst#661](https://github.com/QuantEcon/lecture-python.myst/issues/661)
- **Phase 3**: Enable `force_orphan: true` for future deployments (prevents history re-accumulation)

## Next Steps

- [ ] Perform cleanup on each repository
- [ ] Verify websites are still accessible after each cleanup
- [ ] Delete backup branches after 1-2 weeks of verification
- [ ] Proceed to Phase 3 (enable force_orphan)

**Comment by mmcky** [comment]
## Blocked: Waiting on action-backups

Phase 2 is currently blocked pending the rollout of [QuantEcon/action-backups](https://github.com/QuantEcon/action-backups) to create full repository backups before performing the destructive gh-pages history cleanup.

This provides an additional safety layer beyond the `gh-pages-backup` branches and Phase 1 HTML release archives.
